// Copyright 2018 The Chromium OS Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

package arc_proxy;

// TODO(hidehiko,yusukes,keiichiw): Currently this is compatible with udsproxy.
// Redesign the protocol after we have our own host side implementation.

// Represents a file descriptor to be transferred.
message FileDescriptor {
  enum Type {
    REGULAR = 0;  // To satisfy proto compiler. Unsupported yet.
    // 1-3 is skipped as unsupported.
    FIFO_READ = 4;
    FIFO_WRITE = 5;
    SOCKET = 6;
    // TODO(hidehiko,keiichiw): support more file descriptor types.
  };
  Type type = 1;
  uint64 handle = 2;
}

// Represents a data to be transferred between host and guest.
message Message {
  uint64 handle = 1;
  // 2 is skipped as unused.
  bytes data = 3;
  repeated FileDescriptor transferred_fd = 4;
}
